setup/util/util.go

```diff
@@ -0,0 +1,150 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/go-ini/ini"
+	"github.com/lib/pq"
+	"github.com/pganalyze/collector/setup/query"
+)
+
+func UsingLogExplain(section *ini.Section) (bool, error) {
+	k, err := section.GetKey("enable_log_explain")
+	if err != nil {
+		return false, err
+	}
+	return k.Bool()
+}
+
+func ApplyConfigSetting(setting, value string, runner *query.Runner) error {
+	// N.B.: we don't quote the value because in the case of lists (like shared_preload_libraries)
+	// that does not parse the list correctly
+	err := runner.Exec(fmt.Sprintf("ALTER SYSTEM SET %s = %s", setting, value))
+	if err != nil {
+		return fmt.Errorf("failed to apply setting: %s", err)
+	}
+	err = runner.Exec("SELECT pg_reload_conf()")
+	if err != nil {
+		return fmt.Errorf("failed to reload Postgres configuration after applying setting: %s", err)
+	}
+
+	return nil
+}
+
+func ValidateLogMinDurationStatement(ans interface{}) error {
+	ansStr, ok := ans.(string)
+	if !ok {
+		return errors.New("expected string value")
+	}
+	ansNum, err := strconv.Atoi(ansStr)
+	if err != nil {
+		return errors.New("value must be numeric")
+	}
+	if ansNum < 10 && ansNum != -1 {
+		return errors.New("value must be either -1 to disable or 10 or greater")
+	}
+	return nil
+}
+
+var expectedMd5s = map[string]string{
+	"explain":              "7a0a1784d170975d8538d3b8b38c3fad",
+	"get_stat_replication": "066680efec598232c0245477976a2c3d",
+}
```

Comment:
Nice, thats a great idea - I wonder if we could also check that the function is security definer, and is owned by a superuser?

Reply:
Good call. I think for now I can just add that as predicates to the SQL below.